server/src/test/java/org/cloudfoundry/identity/uaa/oauth/jwt/CommonSignerTest.java

```diff
@@ -109,4 +115,32 @@ public void test_mac_signing_options() throws JOSEException, ParseException {
         assertEquals(UaaMacSigner.SUPPORTED_ALGORITHMS, signer.supportedJWSAlgorithms());
         assertNotNull(signer.getJCAContext());
     }
+
+    @Test
+    public void test_nimbus_singing_with_single_aud_value() throws JOSEException, ParseException {
+        // given
+        Map<String, Object> objectMap = Map.of("sub","1234567890", "name", "John Doe", "aud", Arrays.asList("single"));
+        // when
+        CommonSigner signer = new CommonSigner("id", rsaSigningKey, "http://localhost/uaa");
+        assertEquals("RS256", signer.algorithm());
+        assertEquals("id", signer.keyId());
+        JWSHeader header = new JWSHeader.Builder(JWSAlgorithm.RS256).type(JOSEObjectType.JWT).build();
+        JWTClaimsSet claimsSet = JWTClaimsSet.parse(objectMap);
+        SignedJWT resultedJwt = new SignedJWT(header, claimsSet);
+        resultedJwt.sign(signer);
+        String payLoadString = JWTParser.parse(resultedJwt.serialize()).getParsedParts()[1].decodeToString();
+        // then
+        assertThat(payLoadString, Matchers.containsString("\"aud\":\"single\""));
```

Comment:
this plain decode and match is done to ensure type.... if we could parse it with own Claims class or JWTClaimsSet then we would get an array...